homeassistant/components/filesize/__init__.py

```diff
@@ -24,6 +25,14 @@ def _check_path(hass: HomeAssistant, path: str) -> None:
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up from a config entry."""
     await hass.async_add_executor_job(_check_path, hass, entry.data[CONF_FILE_PATH])
+
+    path = entry.data[CONF_FILE_PATH]
+    get_path = await hass.async_add_executor_job(pathlib.Path, path)
+    fullpath = str(get_path.absolute())
```

Comment:
Mobile so can't validate this: but need to check if absolute() needs to run in the executor since it might do blocking I/O

Reply:
Looks like it could potentially do I/O so I'm adding it to the executor